R/read_file.R

```diff
@@ -1,64 +1,26 @@
-#' Read files with supported extensions
+#' Read .rda files files with supported extensions
 #'
 #' @param path Character; path to a file storing a single object.
 #'
 #' @examples
-#' supported_extensions()
-#'
 #' # Odd case: Read a file saved as .rds but with extension .rda
 #' path <- tempfile(fileext = ".rda")
 #'
 #' data1 <- data.frame(x = 1)
 #' saveRDS(data1, path)
-#' read_file(path)
+#' read_rda(path)
 #'
 #' # More natural case: .RData saved as .rda
 #' data2 <- data.frame(x = 1)
 #' save(data2, file = path)
-#' read_file(path)
+#' read_rda(path)
```

Comment:
We keep `read_rda()` -- which is more focused than `read_file()` -- until we ensure no .rds file is saved with the extension .rda.

Reply:
Nit: Conceptually I  wish that it tried the "right" thing first (read it as an RDA file), and only try the weird thing (read it as an RDS file) just in case that failed, but in this case I guess it makes sense to try to read as RDS first because that's a clear pass or fail operation and there's no way for both reading as RDA and RDS to work on the same file, so it should be safe.

Reply:
TODO: Happy path first.

Reply:
DONE